types/eslint/eslint-tests.ts

```diff
@@ -0,0 +1,524 @@
+sourceCode.getFirstTokenBetween(AST, AST, { skip: 0 });
+sourceCode.getFirstTokenBetween(AST, AST, t => t.type === 'Identifier');
+sourceCode.getFirstTokenBetween(AST, AST, { filter: t => t.type === 'Identifier' });
+sourceCode.getFirstTokenBetween(AST, AST, { skip: 0, filter: t => t.type === 'Identifier' });
+sourceCode.getFirstTokenBetween(AST, AST, { includeComments: true });
+sourceCode.getFirstTokenBetween(AST, AST, { includeComments: true, skip: 0 });
+sourceCode.getFirstTokenBetween(AST, AST, { includeComments: true, skip: 0, filter: t => t.type === 'Identifier' });
+
+sourceCode.getFirstTokensBetween(AST, AST, 0);
+sourceCode.getFirstTokensBetween(AST, AST, { count: 0 });
+sourceCode.getFirstTokensBetween(AST, AST, t => t.type === 'Identifier');
+sourceCode.getFirstTokensBetween(AST, AST, { filter: t => t.type === 'Identifier' });
+sourceCode.getFirstTokensBetween(AST, AST, { count: 0, filter: t => t.type === 'Identifier' });
+sourceCode.getFirstTokensBetween(AST, AST, { includeComments: true });
+sourceCode.getFirstTokensBetween(AST, AST, { includeComments: true, count: 0 });
+sourceCode.getFirstTokensBetween(AST, AST, { includeComments: true, count: 0, filter: t => t.type === 'Identifier' });
+
+sourceCode.getLastTokenBetween(AST, AST, 0);
+sourceCode.getLastTokenBetween(AST, AST, { skip: 0 });
+sourceCode.getLastTokenBetween(AST, AST, t => t.type === 'Identifier');
+sourceCode.getLastTokenBetween(AST, AST, { filter: t => t.type === 'Identifier' });
+sourceCode.getLastTokenBetween(AST, AST, { skip: 0, filter: t => t.type === 'Identifier' });
+sourceCode.getLastTokenBetween(AST, AST, { includeComments: true });
+sourceCode.getLastTokenBetween(AST, AST, { includeComments: true, skip: 0 });
+sourceCode.getLastTokenBetween(AST, AST, { includeComments: true, skip: 0, filter: t => t.type === 'Identifier' });
+
+sourceCode.getLastTokensBetween(AST, AST, 0);
+sourceCode.getLastTokensBetween(AST, AST, { count: 0 });
+sourceCode.getLastTokensBetween(AST, AST, t => t.type === 'Identifier');
+sourceCode.getLastTokensBetween(AST, AST, { filter: t => t.type === 'Identifier' });
+sourceCode.getLastTokensBetween(AST, AST, { count: 0, filter: t => t.type === 'Identifier' });
+sourceCode.getLastTokensBetween(AST, AST, { includeComments: true });
+sourceCode.getLastTokensBetween(AST, AST, { includeComments: true, count: 0 });
+sourceCode.getLastTokensBetween(AST, AST, { includeComments: true, count: 0, filter: t => t.type === 'Identifier' });
+
+sourceCode.getTokensBetween(AST, AST, 0);
+
+sourceCode.getTokens(AST);
+sourceCode.getTokens(AST, 0);
+sourceCode.getTokens(AST, 0, 0);
+sourceCode.getTokens(AST, t => t.type === 'Identifier');
+sourceCode.getTokens(AST, { filter: t => t.type === 'Identifier' });
+sourceCode.getTokens(AST, { includeComments: true });
+sourceCode.getTokens(AST, { includeComments: true, filter: t => t.type === 'Identifier' });
+
+// TODO: Is it token or Node ?
+sourceCode.commentsExistBetween(AST, AST);
```

Comment:
Yes, node, token, or comment.
In fact, it's `{ range: [number, number] }`.

Reply:
And since TS’s typing is structural, we could specify that the arguments are of type `{ range: [number, number] }` and anything that conforms to that could be passed in.

Reply:
@j-f1 I like this approach. What do you think of a `Rangeable` type?

```ts
interface Rangeable {
	range: [number, number]
}
```

It appears that `getFirstToken`, `getLastToken`, `getTokenBefore`, `getTokenAfter`, `getFirstTokenBetween`, `getLastTokenBetween`, `getTokenOrCommentBefore`, `getTokenOrCommentAfter`, `getFirstTokens`, `getLastTokens`, `getTokensBefore`, `getTokensAfter`, `getFirstTokensBetween`, `getLastTokensBetween`, `getTokens`, `getTokensBetween`, `commentsExistBetween`, `getCommentsBefore`, `getCommentsAfter`, `getCommentsInside` can all beneficiate this type.

Reply:
`Ranged`?